src/index.js

```diff
@@ -0,0 +1,221 @@
+      this.adapters.set(adapterName, '');
+      instance[adapterName] = adapterInstance;
+    }
+  }
+
+  /**
+    @function unbindAdapter
+    Remove adapter from the exposed module instance
+    @param adapterName
+  **/
+  unbindAdapter(adapterName) {
+    assert(adapterName);
+    if (this.adapters.has(adapterName)) {
+      delete instance[adapterName];
+      this.adapters.delete(adapterName);
+    }
+  }
+}
+
+/**
+ * Configure and return a chain of processing a log message. It should be required on module launch.
+ * It sets up loggers, notifiers for bugs, and exposes the internal providers, abstracted.
+ * If no parameter is provided, it will return already configured providers
+ * @param  {Object}  [config] Configuration object required by the internal providers
+ * @param  {Array}   [config.chainLinks] Array of instances of chainLinks
+ * @param  {Object}  [config.adapters] Object with adapters to be exposed
+ * @param  {boolean} [config.CONSOLE_LOGGING] Switches on/off the console logging chain
+ * @param  {boolean} [config.MIN_LOG_LEVEL] Minimal log level for a message to be processed
+ * @param  {boolean} [config.MIN_LOG_LEVEL_CONSOLE] Minimal log level for a message to be processed by console chain link (more prior over MIN_LOG_LEVEL)
+ * @param  {boolean} [config.DEFAULT_LOG_LEVEL] Fall back if a message does not have one
+ * Note that this module can also be configured with environment variables.
+ * Environment variables have higher priority over a settings object
+ * @return {Object}  The configured providers
+ */
+module.exports = (config) => {
+  if (!config) {
+    if (instance) return instance;
+    console.error('Logtify should be initilised before used without config.');
+  }
+  const settings = Object.assign({}, config);
+
+  const customChainLinks = settings.chainLinks;
+  const adapters = settings.adapters;
+
+  delete settings.chainLinks;
+  delete settings.adapters;
+
+  const chain = new LoggerChain(settings);
+  instance = { chain };
+
+  // default
+  const chainLinkIndex = chain.push(new ConsoleChainLink(settings, new ChainLinkUtility()));
+  chain.bindAdapter('logger', new Winston(chain, chainLinkIndex));
+
+  // presets
+  if (Array.isArray(settings.presets)) {
+    presets(settings.presets);
+  }
+
+  // custom
+  if (Array.isArray(customChainLinks)) {
+    for (const CustomChainLink of customChainLinks) {
+      if (![null, undefined].includes(CustomChainLink)) {
+        // if constructor
+        if (typeof CustomChainLink === 'function') {
+          chain.push(new CustomChainLink(settings, new ChainLinkUtility()));
+        } else if (typeof CustomChainLink === 'object') {
+          const chainLinkConfig = CustomChainLink.config || settings;
+          const ChainLinkClass = CustomChainLink.class;
+          chain.push(new ChainLinkClass(chainLinkConfig, new ChainLinkUtility()));
+        }
+      }
+    }
+  }
+
+  if (adapters !== null && typeof adapters === 'object') {
```

Comment:
typeof adapters === 'object' on it's own is enough no need fro null check as this check will do it also.